pandas/core/generic.py

```diff
@@ -1986,7 +1986,13 @@ def empty(self) -> bool_t:
     __array_priority__: int = 1000
 
     def __array__(self, dtype: npt.DTypeLike | None = None) -> np.ndarray:
-        return np.asarray(self._values, dtype=dtype)
+        values = self._values
+        arr = np.asarray(values, dtype=dtype)
+        if arr is values and using_copy_on_write():
```

Comment:
Not for this PR: Could we use the same helper as in astype here?

Reply:
I don't think that is needed here since this explicitly only deals with numpy arrays. And numpy.asarray seems to guarantee it either returns the original array or either a copy.

Reply:
On second thought, I missed that `values` isn't necessarily already a numpy array, of course. So you are right, for extension dtypes we need something smarter (that's my todo comment just below, I see now), and then the same helper for astype could indeed help here.

Reply:
Makes sense, thx